### ISSUE TITLE
Rotation of portrait 90 degrees causes clipping

### ISSUE THREAD

**Issue by jwalling** [OpenShot/openshot-qt#1281]
Computer: VAW70 Laptop
https://ctl.net/blogs/news/introducing-the-ctl-vaw70-14-ultra-light-class-notebook
Memory 15.6 GiB
Processor Intel Core i5 4200U 1.60 Ghz x4
Graphics Intel Haswell Mobile
OS type 64 Bit
Disk 487 GB SSD
* Operating System / Distro: ?  Ubuntu 16.04
* OpenShot Version: ? 2.4.1
* _Please attach log files if crash_   >>>> No crash
**Issue Description and steps to reproduce:**
MP4 Video shot in portrait mode
Video loads 90 degrees right
Rotated video 90 degrees left
Top and bottom of video are cropped -- making the video unusable.
I searched the web and the issues and could not find a fix.
I would like to be able to load the video without it being rotated or rotate without cropping.
Kdenlive loads portrait videos correctly.
Renderings:
HDV 720 24p (1280 x 720) cropped - not usable
Mobile 360p (320 x 240) not cropped as much - could use in a pinch

**Comment by peanutbutterandcrackers** [comment]
@DylanC - I had that same issue before. So I guess I can confirm this bug.

@jwalling - For now, as a workaround, please use ffmpeg to resolve this issue.
$ ffmpeg -i /path/to/input.mp4 output.mp4

It will automatically rotate the video back to portrait mode.

Hope that helps, for now...

**Comment by jwalling** [comment]
Thanks. That helped. 
(I tried  ffmpeg before with "transpose=2" and if turned the video 180 degrees. Doh!)
It was supposed to turn 90 degrees CCW.)
One "benefit" of using ffmpeg is the file size shrank from 4GB to 219MB.
I must have lost resolution or frames in translation, but its not an issue so far.

**Comment by DylanC** [comment]
@jwalling - I'm going to leave this issue open for now. I know the workaround helped you but I think its still a bug in OpenShot we should solve.

**Comment by jwalling** [comment]
Oops. There is something wrong with my installation of OpenShot.
Rendering produces blank audio/screen whether I use the raw file or the one translated by ffmpeg.
Back to the drawing board.

**Comment by peanutbutterandcrackers** [comment]
@DylanC - You're the man, boss!

@jwalling - Hmm... that is definitely strange. And you're using Linux too.... Perhaps open another issue and put in the logs and everything. BTW, glad I could be of some assistance. :)

**Comment by N3WWN** [comment]
Can this issue be mitigated by using the `Scale` property of the clip on the timeline?  Try the various `Crop`, `Best Fit`, `Stretch` and `None` options and see if any of them resolve the problem. 

Sorry that I can't test this since I don't have a video on hand that exhibits the issue.

**Comment by peanutbutterandcrackers** [comment]
I think I have tried using the 'Scale' property but without any proper results. I can't remember for certain though...

@jwalling - Could you please try using the method described above. If you're not sure how to change the 'Scale' property: Right click on the clip on the timeline > Click Properties. Now, in the properties tab that opens to the left of the timeline, scroll down to 'scale' and change the value as pointed in the previous comment. Good luck!

**Comment by jwalling** [comment]
@peanutbutterandcrackers @N3WWN 

**Using the raw video.**
Rotation + Scale properties (Crop, Best Fit, Stretch) reduced the cropping top and bottom, but if the subject's face is not centered, the cropping can still remove the top of the subject's head.  That can be useful information for future video capture if someone needs to use portrait.
I used Best Fit plus Scale factors X=Y=0.60 to bring the subjects' heads and bodies back into the video preview frame.
When I rendered a clip of the raw video, it came out rotated 90 degrees clockwise -- Not ideal!!!
So the rotation only affected the preview. Doh!

**Using the video with ffmpeg conversion.**
I didn't use scaling. Rendering a clip came out with correct orientation.
(For some reason rendering failed last night, but is working tonight.)

**Bottom line:** 
Rotation with scaling didn't work.  
Conversion with ffmpeg did work.

**Comment by peanutbutterandcrackers** [comment]
Hmm... I see. So, it's a bug, for real, that needs real attention...

Please do continue to support OpenShot, good sir! :)

**Comment by SamStephens** [comment]
Confirmed that the Scale property appears to use the dimensions of the rotated clip before rotation. This is a problem for Portrait video from some mobile phones, as that video is recorded in Landscape with rotation set to 90 degrees.

**Comment by titsitits** [comment]
I have the same bug with my portrait photos. Any news about this thread ?

**Comment by jaronwakley** [comment]
A work around that worked great for me for multiple images at a time:
1. Open the directory of images in Rawtherapee (I used version 5.8)
2. Select all the images in the file browser within Rawtherapee.
3. Right click and select "Put to queue (Fast export)"
4. Select "Queue" Tab (Top left corner)
5. Export them here.
After exporting the images, they should work.

**Comment by SamStephens** [comment]
I've confirmed the issue is fixed in OpenShot-v2.5.1-dev3-daily. I found a separate issue whilst verifying this fix though: https://github.com/OpenShot/openshot-qt/issues/4207

**Comment by JacksonRG** [comment]
@SamStephens  Thanks for your help! I'll close this one, and take a look at that new issue.
